File: CSAdmin/Clases/SampleIPrincipal.vb
Public Class SampleIPrincipal
    Implements System.Security.Principal.IPrincipal

    Private identityValue As SampleIIdentity

    Public ReadOnly Property Identity() As System.Security.Principal.IIdentity Implements System.Security.Principal.IPrincipal.Identity
        Get
            Return identityValue
        End Get
    End Property

    Public Function IsInRole(ByVal role As String) As Boolean Implements System.Security.Principal.IPrincipal.IsInRole
        Return role = identityValue.Role.ToString
    End Function

    Public Sub New(ByVal name As String, ByVal pass As String)
        identityValue = New SampleIIdentity(name, pass)
    End Sub
End Class

File: CSAdminApp/Clases/SampleIPrincipal.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSAdminApp.Clases
{
    class SampleIPrincipal : System.Security.Principal.IPrincipal
    {
        #region Miembros de IPrincipal

        public System.Security.Principal.IIdentity Identity
        {
            get { throw new NotImplementedException(); }
        }

        public bool IsInRole(string role)
        {
            throw new NotImplementedException();
        }

        #endregion

        public SampleIPrincipal()
        {
        }
    }
}
